src/badges.js

```diff
@@ -0,0 +1,953 @@
+				break;
+		}
+	});
+	if (invalidParam.length) {
+		console.log(chalk.gray(template.path));
+		console.log(chalk.red(`Error: at Template ${template.details?.name ? `'${template.details?.name}' ` : ''}has the following issues:`));
+		invalidParam.forEach((param) => {
+			console.log('\t - ' + chalk.cyanBright(`${param}`));
+		});
+		exit(1);
+	}
+}
+
+export async function syncBadgeTemplate(options) {
+	const { context } = options;
+	const { searchspring, repository } = context;
+	const [command, templateName] = options.args;
+	const { secretKey } = options.options;
+
+	if (!searchspring || !context.project || !context.project.path) {
+		console.log(chalk.red(`Error: No Snap project found in ${process.cwd()}.`));
+		return;
+	}
+
+	const locations = await getLocationsFile(context.project.path);
+	if (locations) {
+		validateLocations(locations);
+	}
+
+	const templates = await getTemplates(context.project.path);
+	const syncTemplates = templates.filter((template) => {
+		validateTemplate(template, locations);
+		if (templateName && template.details.name == templateName) {
+			return template;
+		}
+		return template;
+	});
+
+	if (!syncTemplates.length) {
+		console.log(chalk.red(`Error: Template(s) not found.`));
+		return;
+	}
+
+	const sync = async (template, secretKey) => {
+		const payload = buildBadgeTemplatePayload(template.details);
+		const remoteTemplates = await new ConfigApi(secretKey, options.dev).getBadgeTemplates();
+		const remoteTemplate = remoteTemplates.badgeTemplates?.find((remoteTemplate) => remoteTemplate.tag == template.details.name);
+		let skipTemplateUpdate = false;
+		if (remoteTemplate) {
+			try {
+				// check if remote template already matches local template
+				const { name, description, snapComponent, labelConfig, locations } = remoteTemplate;
+				// object property order stored in db may differ than what's synced, map to match payload order for comparison below
+				const properties = JSON.parse(remoteTemplate.properties).map((remoteParameter) => {
+					return {
+						name: remoteParameter.name,
+						type: remoteParameter.type,
+						label: remoteParameter.label,
+						description: remoteParameter.description,
+						defaultValue: remoteParameter.defaultValue,
+						validations: remoteParameter.validations,
+						options: remoteParameter.options,
+					};
+				});
+				const templateMatchRemote =
+					payload.label === name &&
+					payload.description === description &&
+					payload.component === snapComponent &&
+					JSON.stringify(payload.value) === JSON.stringify(JSON.parse(labelConfig)) &&
+					JSON.stringify(payload.locations.sort()) === JSON.stringify(JSON.parse(locations).sort()) &&
+					JSON.stringify(payload.parameters) === JSON.stringify(properties);
+
+				if (templateMatchRemote) {
+					console.log(
+						chalk.green(
+							`        ${template.details.name} - ${chalk.yellow(`remote template matches local template payload. This template will not be synced`)}`
```

Comment:
Maybe instead of "remote template matches local template payload. This template will not be synced" - just "no changes to sync" or something a little less verbose?